src/Rings/integer-localizations.jl

```diff
@@ -0,0 +1,197 @@
+  end
+end
+
+### required getter functions
+ambient_ring(S::FmpzComplementOfZeroIdeal) = ZZ
+
+### required functionality
+function Base.in(b::fmpz, S::FmpzComplementOfZeroIdeal) 
+  return !(b == zero(ZZ))
+end
+
+
+#################################################################################
+# Localizations of ℤ                                                            #
+#################################################################################
+
+@Markdown.doc """
+    FmpzLocalizedRing{MultSetType} <: AbsLocalizedRing{FlintIntegerRing, fmpz, MultSetType} 
+
+A minimal implementation for the localization `ℤ[S⁻¹]` of the ring of integers 
+at a multiplicatively closed set `S` of type `MultSetType`.
+"""
+mutable struct FmpzLocalizedRing{MultSetType} <: AbsLocalizedRing{FlintIntegerRing, fmpz, MultSetType} 
+  S::MultSetType # The multiplicatively closed set S ⊂ R whose inverses are added to R
+
+  function FmpzLocalizedRing(S::MultSetType) where {MultSetType}
+    # Sanity check whether the multiplicatively closed set S is compatible with the 
+    # given rings
+    MultSetType <: AbsMultSet{FlintIntegerRing, fmpz} || error(
+	"The type of the multiplicatively closed set is not compatible with the type of the ring"
+	)
+    return new{MultSetType}(S)
+  end
+end
+
+### required getter functions
+original_ring(W::FmpzLocalizedRing{MultSetType}) where {MultSetType} = ZZ::FlintIntegerRing
+inverted_set(W::FmpzLocalizedRing{MultSetType}) where {MultSetType} = W.S::MultSetType
+
+### required extensions of the localization function
+localize_at(S::FmpzComplementOfPrimeIdeal) = FmpzLocalizedRing(S)
+localize_at(S::FmpzComplementOfZeroIdeal) = FmpzLocalizedRing(S)
+localize_at(S::FmpzPowersOfElement) = FmpzLocalizedRing(S)
+
+
+#######################################################################
+# Elements of localizations of ℤ                                      #
+#######################################################################
+
+@Markdown.doc """
+    FmpzLocalizedRingElem{MultSetType}
+
+Elements `a/b ∈ ℤ[S⁻¹]` of localizations of the ring of integers 
+at a multiplicatively closed set `S` of type `MultSetType`.
+"""
+mutable struct FmpzLocalizedRingElem{MultSetType} <: AbsLocalizedRingElem{FlintIntegerRing, fmpz, MultSetType}
+  numerator::fmpz
+  denominator::fmpz
+  R::FmpzLocalizedRing{MultSetType} # the parent ring
+
+  function FmpzLocalizedRingElem(R::FmpzLocalizedRing{MultSetType}, a::fmpz, b::fmpz) where {MultSetType}
+    # Perform some sanity checks
+    b in inverted_set(R) || error("$b does not belong to the units of $R")
+    return new{MultSetType}(a, b, R)
+  end
+end
+
+### required getter functions
+parent(f::FmpzLocalizedRingElem{MultSetType}) where {MultSetType} = f.R::FmpzLocalizedRing{MultSetType}
+numerator(f::FmpzLocalizedRingElem{MultSetType}) where {MultSetType} = f.numerator::fmpz
+denominator(f::FmpzLocalizedRingElem{MultSetType}) where {MultSetType} = f.denominator::fmpz
```

Comment:
```suggestion
numerator(f::FmpzLocalizedRingElem) = f.numerator::fmpz
denominator(f::FmpzLocalizedRingElem) = f.denominator::fmpz
```
Also there are no tests for these three functions, as the Codecov comment indicates (you are probably already aware, though)

Reply:
There are no tests for either `integer-localization.jl` nor for `nmod-localizations.jl`. Since the functionality of these files is not crucial, I didn't care to write a proper test file. Do you think this is necessary?